platform/linux-generic/include/odp_chksum_internal.h

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2013, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+/**
+ * @file
+ *
+ * ODP checksum - implementation internal
+ */
+
+#ifndef ODP_CHKSUM_INTERNAL_H_
+#define ODP_CHKSUM_INTERNAL_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Simple implementation of ones complement sum.
+ * Based on RFC1071 and its errata.
+ */
+static uint32_t _odp_chksum_ones_comp16_32(const void *p, uint32_t len,
+					   odp_bool_t odd_offset)
+
+{
+	uint32_t sum = 0;
+	const uint16_t *data = p;
+
+	if (odd_offset) {
+		uint16_t left_over = 0;
+
+		*(uint8_t *)&left_over = *(const uint8_t *)data;
+		sum = left_over;
+		data++;
```

Comment:
This erroneously increments `data` by 2 when you want to increment it by 1 here so that it points to `p + 1`, not `p + 2`. Correct is `data = (const uint16_t *)((uintptr_t)p + 1);`

Reply:
This still needs fixing.